documentation/testing/integration-tests.md

```diff
@@ -0,0 +1,201 @@
+    }    
+
+    // Use internal testing constructor
+    private val node = SomeScreenBuilder(
+        dependency = object : SomeScreen.Dependency {
+            override val network: Network = testNetwork
+            override val analyticsTracker: AnalyticsTracker = testTracker
+        },
+        childBuilders = object : SomeScreenBuilders {
+            override val child1 = child1Builder
+        }
+    ).build(
+        buildContext = BuildContext.root(
+            savedInstanceState = null,
+            // Override view implementation with stub
+            customisations = SomeScreen.Customisation { { view } }.toDirectory()
+        )
+    )
+
+    ...
+
+    @Test
+    fun `GIVEN Child1 is attached WHEN Child1  THEN output SomeOutput`() {
+        nodeTestHelper.moveToStateAndCheck(Lifecycle.State.STARTED) {
+            val test = child1Builder.last?.input?.test()
+
+            // Fake Child1 ouput
+            child1Builder.last?.output?.accept(Child1.Output.SomeThing)
+
+            // Verify Child1 input
+            test.assertValue(Child1.Input.SomeThing)
+
+            // Verify Child1 was dettached and destroyed
+            child1Builder.assertLastNodeState(Lifecycle.State.DESTROYED)
+        }
+    }
+
+}
+```
+
+## UI Tests
+
+The framework also provides a way to write UI integration tests for `Node` implementation.
+You can use `RibsRule` to write the same test as above, but with real view implementation.
+You are free to use both fake and real children to test different levels of integration.
+When you fake children, then use `RibBuilderStub` to send inputs and verify outputs.
+When you use real children, then use `Espresso` and stubs to move both children and parent into desired state.
+
+```kotlin
+class SomeScreenTest {
+
+    @get:Rule
+    val ribsRule = RibsRule()
+
+    // Send inputs to the Rib
+    private val input = PublishRelay.create<EmptyScreen.Input>()
+    // Verify outputs from the Rib
+    private val output = PublishRelay.create<EmptyScreen.Output>()
+
+    private fun buildRib(activity: RibTestActivity, bundle: Bundle?) =
+        // We do not replace any children in this test, but you are free to do it
+        SomeScreenBuilder(
+            dependency = object : SomeScreen.Dependency {
+                override val network: Network = ...
+                override val analyticsTracker: AnalyticsTracker = ...
+            },
+        ).build(root(bundle)).also {
+            // Wire inputs and outputs
+            input.subscribe(it.input)
+            it.output.subscribe(output)
+        }
+
+    @Before
+    fun before() {
+        ribsRule.start { activity, bundle -> buildRib(activity, bundle) }
+    }
+
+    @Test
+    fun WHEN_press_refresh_button_THEN_send_SomeOuput() {
+        val outputTest = output.test()
+
+        onView(withId(R.id.title)).perform(click())
+
+        outputTest.assertValue(SomeScreen.Output.SomeOutput)
+    }
+
+}
+```
+
+# Compatibility with other frameworks
```

Comment:
This part is literally the same as the Compatibility page. Should we just remove it and/or point to the other page?